pczt/src/roles/updater/transparent.rs

```diff
@@ -0,0 +1,42 @@
+use zcash_primitives::transaction::components::transparent::pczt::{
+    ParseError, Updater, UpdaterError,
+};
+
+use crate::Pczt;
+
+impl super::Updater {
+    /// Updates the transparent bundle with information in the given closure.
+    pub fn update_transparent_with<F>(self, f: F) -> Result<Self, TransparentError>
+    where
+        F: FnOnce(Updater<'_>) -> Result<(), UpdaterError>,
+    {
+        let Pczt {
+            global,
+            transparent,
+            sapling,
+            orchard,
+        } = self.pczt;
+
+        let mut bundle = transparent
+            .into_parsed()
+            .map_err(TransparentError::Parser)?;
+
+        bundle.update_with(f).map_err(TransparentError::Updater)?;
```

Comment:
This doesn't check the modifiability flags.

Reply:
Those are irrelevant here, as the only permitted changes via an Updater are non-effecting.

Reply:
I misread; this cannot cause mutability problems that are not addressed (or not, as the case may be!) by checks elsewhere.